src/Marker.js

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { renderToString } from 'react-dom/server';
+import InfoWindow from './InfoWindow';
+import { MapContext } from './KakaoMap';
+
+class Marker extends React.Component {
+    static contextType = MapContext;
+    constructor(props) {
+        super(props);
+        
+        this.state = {};
+    }
+
+    componentDidMount() {
+        const {
+            latitude,
+            longitude,
+            details,
+            markerImgSrc,
+            markerImgWidth,
+            markerImgHeight
+        } = this.props;
+
+        const marker = new kakao.maps.Marker({
+            position: new kakao.maps.LatLng(
+                latitude,
+                longitude
+            ),
+            image:
+                markerImgSrc && markerImgWidth && markerImgHeight
+                    ? this.createMarkerImg(
+                            markerImgSrc,
+                            markerImgWidth,
+                            markerImgHeight
+                        )
+                    : '',
+        });
+        details
+            ? this.createCustomOverlay(this.context, marker, details)
+            : ''; 
+        marker.setMap(this.context);
+    }
+
+    createMouseClickListener = (map, customOverlay) => {
+        let overlay = false;
+        const toggle = () => (overlay = !overlay);
+        return () => {
+            toggle();
+            overlay ? customOverlay.setMap(map) : customOverlay.setMap(null);
```

Comment:
You can change this code to simple like below.

`customOverlay.setMap(overlay ? map : null);`